VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -191,4 +194,27 @@ public static void setEndorsedState(@NonNull TextView target,
         }
 
     }
+
+    /**
+     * Renders hyperlinks in the given html to clickable items when applied to the given TextView.
+     *
+     * @param textView The textView which will render the given html.
+     * @param html     The HTML to apply.
+     */
+    public static void linkifyTextView(@NonNull TextView textView, @NonNull String html) {
+        Spanned spannedHtml = (Spanned) DiscussionTextUtils.parseHtml(html);
+        Object[] urlSpans = spannedHtml.getSpans(0, spannedHtml.length(), URLSpan.class);
+        textView.setAutoLinkMask(Linkify.ALL);
+        textView.setMovementMethod(LinkMovementMethod.getInstance());
+        textView.setText(spannedHtml);
+
+        SpannableString tvText = (SpannableString) textView.getText();
+        for (Object spanObj : urlSpans) {
+            URLSpan span = (URLSpan) spanObj;
+            int flags = spannedHtml.getSpanFlags(span);
+            int start = spannedHtml.getSpanStart(span);
+            int end = spannedHtml.getSpanEnd(span);
```

Comment:
OK, then you should move the `flags` variable at the end to at least define them in the same order as the parameters.